### PR TITLE
Fabio Cuomo - UI Design Patterns Challenge

### PR DESCRIPTION
**Pull request by mulp**
I had trouble adding a new language, for some reason XCode didn't let me add it using the standard way (the finish button was disabled all the time). I ended up by adding it manually, tests are passing so I shouldn't have broken anything. Just in case let me know. Thanks!

### CODE REVIEW

**Review by caiozullo — COMMENTED** [review]
Hi @mulp, you almost passed the challenge! 

Your implementation looks great, but the `Greek (el)` localization is not working anymore. Looks like adding the locale manually caused some issues. I added some comments that might help.

I also left some other minor suggestions about the implementation.

Please, feel free to push changes, and let me know once you're done so we can review it again. Thanks! 🙂

File: MVP Module/Source/Feed Presentation/FeedPresenter.swift [hunk 1]
```diff
@@ -33,5 +39,6 @@ final class FeedPresenter {
 	
 	func didFinishLoadingFeed(with error: Error) {
 		loadingView.display(FeedLoadingViewModel(isLoading: false))
+        loadingErrorView.display(FeedLoadingErrorViewModel(errorMessage: Localized.Feed.loadError))
```
[the remark this answers is not in the document]
Comment (mulp) [reply MVP Module/Source/Feed Presentation/FeedPresenter.swift:42]: Thank you

File: MVVM Module/Source/Feed Presentation/FeedViewModel.swift [hunk 2]
```diff
@@ -20,13 +20,17 @@ final class FeedViewModel {
...
 		onLoadingStateChange?(true)
 		feedLoader.load { [weak self] result in
-			if let feed = try? result.get() {
-				self?.onFeedLoad?(feed)
-			}
+            switch result {
+            case let .success(feed):
+                self?.onFeedLoad?(feed)
+            case let .failure(error):
+                self?.onFeedLoadError?(error)
```
[the remark this answers is not in the document]
Comment (mulp) [reply MVVM Module/Source/Feed Presentation/FeedViewModel.swift:32]: I don't actually need the localised error message, in the error case, I present the localised FEED_VIEW_CONNECTION_ERROR message

File: UIDesignPatternsChallenge.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -334,8 +335,12 @@
 		08F3B6D8232258BD00510CE7 /* FeedLocalizationTests.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = FeedLocalizationTests.swift; sourceTree = "<group>"; };
 		08F3B6E5232258CE00510CE7 /* XCTestCase+MemoryLeakTracking.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = "XCTestCase+MemoryLeakTracking.swift"; sourceTree = "<group>"; };
 		08F9A7B82534632300B1568F /* en */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = en; path = en.lproj/LaunchScreen.strings; sourceTree = "<group>"; };
-		08F9A7BA2534635300B1568F /* el */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = el; path = el.lproj/LaunchScreen.strings; sourceTree = "<group>"; };
+		08F9A7BA2534635300B1568F /* en */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = en; path = it.lproj/LaunchScreen.strings; sourceTree = "<group>"; };
```
[the remark this answers is not in the document]
Comment (mulp) [reply UIDesignPatternsChallenge.xcodeproj/project.pbxproj:338]: 😞

File: MVVM Module/Source/Feed Presentation/FeedViewModel.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (caiozullo) [reply MVVM Module/Source/Feed Presentation/FeedViewModel.swift]: That's what I meant 😅

**Review by caiozullo — COMMENTED** [review]
Hi @mulp,

Congratulations, you've passed the challenge 🥇

- All tests are passing ✅
- Your code is clean and handles edge cases ✅
- The UI and animations look and behave as expected ✅
- Light and Dark mode support ✅
- Added an extra language ✅

Great job!
